smart_card_connector_app/src/reader-tracker.js

```diff
@@ -0,0 +1,181 @@
+/** @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file contains the class used to track readers and their
+ * status based on messages received from the hook setup on PC/SC code inside
+ * NaCl (check readerfactory_nacl.cc for more information).
+ */
+
+goog.provide('GoogleSmartCard.ReaderInfo')
+goog.provide('GoogleSmartCard.ReaderTracker');
+
+goog.require('GoogleSmartCard.DebugDump');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.TypedMessage');
+goog.require('goog.asserts');
+goog.require('goog.json.NativeJsonProcessor');
+goog.require('goog.log.Logger');
+goog.require('goog.messaging.AbstractChannel');
+goog.require('goog.object');
+goog.require('goog.structs.Map');
+
+goog.scope(function() {
+
+/** @const */
+var GSC = GoogleSmartCard;
+
+/**
+ * Structure used to store information about the reader.
+ * @param {string} name
+ * @param {string=} status
+ * @param {string=} error
+ * @constructor
+ * @struct
+ */
+GSC.ReaderInfo = function(name, status, error) {
+  this.name = name;
```

Comment:
I don't see why it is safe.
Look at this example, where the compiler renames properties "foo" and "bar" into "b" and "a":
https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540formatting%2520pretty_print%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%2520%2540struct%2520%2540constructor%2520*%252F%250Afunction%2520A()%2520%257B%250A%2520%2520this.foo%2520%253D%25201%253B%250A%2520%2520this.bar%2520%253D%25202%253B%250A%257D%250A%250Aa.foo%2520%253D%25203%253B%250Aa.bar%2520%253D%25204%253B%250A%250Aconsole.log(a)%253B
There is no guarantee that the compiler will do the renaming in the same way while compiling several files. So when you'll pass this object from one window to another, there's a chance that the properties would be named in a different manner than it was expected in the target script.